LLVM 17 tblgen fails to build with some C++ errors:
```
In file included from /build/llvm-tblgen-src-17.0.6/clang/include/clang/Basic/TokenKinds.h:51:
/build/llvm-tblgen-src-17.0.6/clang/include/clang/Basic/TokenKinds.def:23:26: error: pasting "kw_" and "[" does not give a valid preprocessing token
   23 | #define KEYWORD(X,Y) TOK(kw_ ## X)
      |                          ^~~
/build/llvm-tblgen-src-17.0.6/clang/include/clang/Basic/TokenKinds.def:756:30: note: in expansion of macro 'KEYWORD'
  756 | #define KEYWORD_ATTRIBUTE(X) KEYWORD(X, KEYALL)
      |                              ^~~~~~~
/build/llvm-tblgen-src-17.0.6/llvm/build/tools/clang/include/clang/Basic/AttrTokenKinds.inc:9:1: note: in expansion of macro 'KEYWORD_ATTRIBUTE'
    9 | KEYWORD_ATTRIBUTE(__arm_streaming)
      | ^~~~~~~~~~~~~~~~~
/build/llvm-tblgen-src-17.0.6/llvm/build/tools/clang/include/clang/Basic/AttrTokenKinds.inc:9:19: warning: 'streaming' attribute only applies to function types [-Wattributes]
    9 | KEYWORD_ATTRIBUTE(__arm_streaming)
      |                   ^~~~~~~~~~~~~~~
/build/llvm-tblgen-src-17.0.6/clang/include/clang/Basic/TokenKinds.h:26:16: note: in definition of macro 'TOK'
   26 | #define TOK(X) X,
      |                ^
/build/llvm-tblgen-src-17.0.6/clang/include/clang/Basic/TokenKinds.def:756:30: note: in expansion of macro 'KEYWORD'
  756 | #define KEYWORD_ATTRIBUTE(X) KEYWORD(X, KEYALL)
      |                              ^~~~~~~
/build/llvm-tblgen-src-17.0.6/llvm/build/tools/clang/include/clang/Basic/AttrTokenKinds.inc:9:1: note: in expansion of macro 'KEYWORD_ATTRIBUTE'
    9 | KEYWORD_ATTRIBUTE(__arm_streaming)
      | ^~~~~~~~~~~~~~~~~
/build/llvm-tblgen-src-17.0.6/llvm/build/tools/clang/include/clang/Basic/AttrTokenKinds.inc: In function 'constexpr bool clang::tok::isRegularKeywordAttribute(TokenKind)':
/build/llvm-tblgen-src-17.0.6/clang/include/clang/Basic/TokenKinds.h:112:44: error: 'kw___arm_streaming' is not a member of 'clang::tok'
  112 | #define KEYWORD_ATTRIBUTE(X) || (K == tok::kw_##X)
      |                                            ^~~
/build/llvm-tblgen-src-17.0.6/llvm/build/tools/clang/include/clang/Basic/AttrTokenKinds.inc:9:1: note: in expansion of macro 'KEYWORD_ATTRIBUTE'
    9 | KEYWORD_ATTRIBUTE(__arm_streaming)
      | ^~~~~~~~~~~~~~~~~
```

--- a/pkgs/development/compilers/llvm/common/lldb.nix
+++ b/pkgs/development/compilers/llvm/common/lldb.nix
@@ -54,9 +54,10 @@ stdenv.mkDerivation (rec {
   src = src';
   inherit patches;
 
-  # LLDB expects to find the path to `bin` relative to `lib` on Darwin. It can’t be patched with the location of
-  # the `lib` output because that would create a cycle between it and the `out` output.
-  outputs = [ "out" "dev" ] ++ lib.optionals (!stdenv.hostPlatform.isDarwin) [ "lib" ];
+  # There is no `lib` output because some of the files in `$out/lib` depend on files in `$out/bin`.
+  # For example, `$out/lib/python3.12/site-packages/lldb/lldb-argdumper` is a symlink to `$out/bin/lldb-argdumper`.
+  # Also, LLDB expects to find the path to `bin` relative to `lib` on Darwin.
+  outputs = [ "out" "dev" ];
 
   sourceRoot = lib.optional (lib.versionAtLeast release_version "13") "${src.name}/${pname}";
 
@@ -140,7 +141,7 @@ stdenv.mkDerivation (rec {
   '';
 
   postInstall = ''
-    wrapProgram $out/bin/lldb --prefix PYTHONPATH : $lib/${python3.sitePackages}/
+    wrapProgram $out/bin/lldb --prefix PYTHONPATH : ''${!outputLib}/${python3.sitePackages}/
 
     # Editor support
     # vscode:

